### ISSUE TITLE
Save email as file: Incorrect parsing of subject creates garbled file name

### ISSUE THREAD

**Issue by LittleAlf** [RealRaven2000/FiltaQuilla#53]
First of all, thanks for your great work of transferring this wonderful addon to Thunderbird 68+! I really appreciate your work of keeping this addon alive, because I use it regularly on quite a number of emails.

**Here is what I found:**
I have several regular emails with logging data and standard subjects. For the first time I would like to save them automatically via FiltaQuilla. As I noticed, the file name of the saved .eml-file **_differs_**, if I save it manually by <right click> "Save as ...", and if I use the "Save as file" filter in FiltaQuilla!

**An example:** 
- The email has the subject "_Solar.web Report für PV-Anlage Muxx (18.01.2020)_" in Thunderbird
- If I save it manually, the proposed file name is "_Solar.web Report für PV-Anlage Maas (18.01.2020).eml_".
- But if I use the filter "Save as ..." from FiltaQuilla, the used name is "_**solarweb-report-utf-8bzso8cg-pv-anlage--muxx-18012020**_".

**As can be seen:**
- All capital letters are changed to lowercase letters
- All spaces are changed to hyphens
- The German word "für" with the German u umlaut, preceeded with an utf-8 modifier is garbled
- The parantheses are removed

**Source text**
When I took a look at the source text of the email, I noticed the subject line to be as follows:
`Subject: Solar.web Report =?utf-8?b?ZsO8cg==?= PV-Anlage  Muxx (18.01.2020)`

It seems that the word _**für**_ is encoded as UTF-8 entirely, and that this is not parsed properly by FiltaQuilla. As a consequence, it seems like the other changes mentioned do occur.

(Unfortunately, I did not find where the subject for this function is parsed. That is why I could not take a first look myself.)

Any improvement is highly appreciated.
LittleAlf

**Comment by raspopov** [comment]
Indeed. Using "Save Message As File" is almost unusable in any other language except English, because of UTF-8 commonly encoded subjects in other languages (Russian included). For example:
```
Subject: =?utf-8?b?0JrQsNGB0YHQvtCy0YvQuSDRh9C10Log0L7RgiDQntCe0J4=?=
 =?utf-8?b?ICLQm9Cw0LHQuNGA0LjQvdGCLtCg0KMi?= 07.08.2021
```
Should be decoded as:
```
Кассовый чек от ООО "Лабиринт.РУ" 07.08.2021
```
but in current version as:
```
utf-8b0jrqsngb0yhqvtcy0yvqusdrh9c10log0l7rgidqntce0j4-utf-8b
```
i.e. un-decoded and without **second** line.

---

P.S. Great extension, I supported it by quickFilters Pro registration!

**Comment by RealRaven2000** [comment]
Just working on this - the first part (decoding the MIME charset) was relatively easy using MailServices.mimeConverter.decodeMimeHeader() but the problem is that there was a sanitization function that removed all non-arabic characters. The only built in function I found so far that does this is in C++ and thus cannot be scripted via JavaScript:

https://searchfox.org/comm-central/source/mailnews/base/src/nsMessenger.cpp#111

I am not too sure what kind of restrictions  apply nowadays for file names, more research is necessary.

**Comment by RealRaven2000** [comment]
I changed the algorithm and only filter out certain characters instead.
1) Replacing the following characters with `-` : `@ | / \`
2) Completely Removing the following characters: `$ % " < > ,`

I tested with the subject [filtaquillasaveas]-кассовый-чек-от-ооо-лабиринт.ру-07.08.2021. The string will be decoded after 60 characters and then the extension ".eml " appended: the file created was named
[filtaquilla-saveas]-кассовый-чек-от-ооо-лабиринт.ру-07.08.2.eml

Test version:
[filtaquilla-3.2pre72.zip](https://github.com/RealRaven2000/FiltaQuilla/files/6975464/filtaquilla-3.2pre72.zip)

**Comment by nomeata** [comment]
These changes made it into the 3.2 already, right? This broke my brother’s setup, because some of the mails he is saving using your plugin have colons (`:`) in the subject, which Windows doesn't like at all…

Could you add `:` to the first list of characters (those that you replace with `-`) for the next release? Thanks!

**Comment by RealRaven2000** [comment]
> These changes made it into the 3.2 already, right? This broke my brother’s setup, because some of the mails he is saving using your plugin have colons (`:`) in the subject, which Windows doesn't like at all…
> 
> Could you add `:` to the first list of characters (those that you replace with `-`) for the next release? Thanks!

yeah, that makes total sense. Somehow I overlooked these. Try this version please?

[filtaquilla-3.2.1pre3.zip](https://github.com/RealRaven2000/FiltaQuilla/files/7061332/filtaquilla-3.2.1pre3.zip)

**Comment by nomeata** [comment]
That works, thanks for the quick fix! :-)
